lib/Documentable/Search.pm6

```diff
@@ -11,32 +12,42 @@ class Documentable::Search {
     method generate-entries($registry) {
         say "Generating entries";
         my @entries;
-        for Kind::Type, Kind::Language, Kind::Programs -> $kind {
-            @entries.append: $registry.lookup($kind.Str, :by<kind>).map(-> $doc {
-                self.search-entry(
-                    category => $doc.subkinds[0],
-                    value    => $doc.name,
-                    url      => $doc.url
-                )
-            }).Slip;
+        self.consume-entries-by-kind(@entries, $registry, Kind::Type, 'Types');
```

Comment:
Suggestion: DRY to a loop with the types and strings. As a matter of fact, the type can be computed via introspection.

Reply:
The explicit code like that, just three lines covering three "main" kinds of documents is IMO easier to read, and just a single call is hardly something to be concerned about being duplicated 3 times.
Can it be written in a simpler way using a loop?